### PR TITLE
incl root flags in auto-completion along with subcommands

### PR DESCRIPTION
**Pull request by awaismslm**
The flag predictor is not working along with complete. Strangely either complete works or predict. But do not work simultaneously.
That is why, I have fixed the issue with `complete`.

### CODE REVIEW

**Comment by awaismslm** [comment]
@UmanShahzad Please review the PR.

**Review by UmanShahzad — CHANGES_REQUESTED** [review]
So this is kind of a hack to the actual solution.

The way autocompletion for flags should work it should only show up when we press `TAB` after inputting a `-`.

You can see this exact behavior with `ipinfo cache` which has a subcommand `clear` but also flags. If you tab, it'll autocomplete `clear`, but if you put a `-` first and then `TAB`, it'll show flags.

**Comment by awaismslm** [comment]
@UmanShahzad It is working fine now as you mentioned above. I did a little tweak in not completed args. Please review,

File: lib/complete/complete.go [hunk 1]
```diff
@@ -133,6 +133,9 @@ reset:
 		return c.suggestLeafCommandOptions(), nil
 	case !arg.Completed:
 		// Currently typing a sub command.
+		if arg.HasFlag {
+			return c.suggestLeafCommandOptions(), nil
```
Comment (UmanShahzad) [inline lib/complete/complete.go:137]: Wouldn't this prioritize showing flags instead of subcommands now? Or does it show both?

Comment (awaismslm) [reply lib/complete/complete.go:137]: @UmanShahzad It checks for the `-` at the start. If it's a flag, only then does it get prioritized.

Here is the current behavior.
If users type without `-` or just press the tab, the following options will show up.

image

if the user starts with `-`, the following options will show up.

image

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**